### ISSUE TITLE
‘’hupan assemSta ‘’  run  too long time  and use too lagre memory

### ISSUE THREAD

**Issue by chenshixuan** [SJTU-CGM/HUPAN#6]
Hi，Dr Duan：
     I used hupan assemSta ，with QUAST 4.5，My genome is about 1G，but ,this step use  too long time  (7days still  mucmer align) and use too lagre memory(350G),do you have some good suggestion to run faster  with fewer time ,Thank you

**Comment by zhqduan** [comment]
Considering the large genome size of human and other eukaryotes, we developed a two-step strategy: discarding the contigs highly similar with the reference genome followed by extracting non-reference sequences. This process is involved in three commands in HUPAN: alignContig, extractSeq and assemSta. Please see the "(4) Extract non-reference sequences from assembled contigs" in [Readme.md ](https://github.com/SJTU-CGM/HUPAN)

**Comment by chenshixuan** [comment]
yes,I have use the  alignContig, extractSeq ,  but this step assemSta  is using too large memory. the quast_result is still running,with the log is ''No handlers could be found for logger "quast" ', and it appered 6times,.and the quast_result/err/sample1.1.err  has a  warning " Reference genome is fragmented. You may consider rerunning QUAST using --fragmented option. QUAST will try to detect misassemblies caused by the fragmentation and mark them fake (will be excluded from # misassemblies)"

**Comment by zhqduan** [comment]
Which reads you used for assembling, short reads or long reads? How many is the contig N50 size? How many sequences left after removing the contigs highly similar with the reference genome? In addition, whether the percentage of repetitive elements of your studied genome is higher? As I know, more repeat sequences will take long time in QUAST 4.5.
The log "No handlers could be found for logger" means that the parameter of multiple threads is not work in QUAST, so your work may only run in one thread. Please check it.
The warning means that the reference genome you used were fragment (you may use a self-constructed reference genome, which contains a lot of contigs or scaffolds). In HUPAN, the misassemblies  is not considered. Therefore, it is not matter of the warnings.

**Comment by chenshixuan** [comment]
My  asssembling contig was use the Nanopore reads，contains   about 6600 contigs,the N50  is  about4M ,the N90 is  47K ( number：2183).  The step2_extractseq  log "There are :5707 contigs in the contigs file.
There are: 39 contigs highly similarity with the reference genome with >= 95% identity and >= 95% coverage .
There are :5668 candidate unaligned contigs.
There are :6805 contigs in the contigs file.
There are: 2 contigs highly similarity with the reference genome with >= 95% identity and >= 95% coverage .
There are :6803 candidate unaligned contigs.
There are :5686 contigs in the contigs file.
There are: 1 contigs highly similarity with the reference genome with >= 95% identity and >= 95% coverage .
There are :5685 candidate unaligned contigs."
 My genome repetitive elementsis abou 65%.
The log "No handlers could be found for logger" means that the parameter of multiple threads is not work in QUAST," it cost about 30   threads at first, one day later use 1thread .
And I  had used a higher version quast witg minimap2, it use one thread  and run very fast(abou 2 days) with a much lower  memory (only 40G ). BUT the result is not suitble for "hupan  getUnalnCtg". So do ypu have plan to update the quast version and the output result to solve my problem?
Thank you very  much!
